[FEATURE] Move issue links to work package controller

hschink opened the pull request:
Implements [ticket 1418](https://www.openproject.org/issues/1418)

Commit:
Removes unused view

Commit:
Implements quoting for work package edit form

Commit:
Uses work package controller's index action

Commit:
Uses work packages path

Commit:
Moves #quoted tests to specs

[skip ci]

Commit:
Reworks quoting spec layout

[skip ci]

Commit:
Adds description quoting cuke

[skip ci]

Commit:
Adds changelog entry

apfelfabrik commented on app/controllers/work_packages_controller.rb:
```diff
@@ -259,6 +273,37 @@ def index
+    respond_to do |format|
+      format.js { render :partial => 'edit', locals: locals }
+      format.html { render :action => 'edit', locals: locals }
+      format.xml  { }
```
Isn't xml disabled by a before_filter?

apfelfabrik commented on app/controllers/work_packages_controller.rb:
```diff
@@ -104,17 +105,29 @@ def new_type
-    safe_params = permitted_params.update_work_package(:project => project)
-    work_package.update_by(current_user, safe_params)
+    if params.has_key? :journal_id
+      journal = Journal.find(params[:id])
```
~~I'm currently having a look whether this might be a problem to use the id w/o typecast, considering MySQL string comparison vulnerabilities earlier this year.~~ No problem here. The problem was w/ strings that are compared to stuff that wrongfully is considered an int.

apfelfabrik commented on app/helpers/journals_helper.rb:
```diff
@@ -61,9 +61,8 @@ def render_journal_details(journal, header_label = :label_updated_time_by, model
-    controller = "/#{model.class.name.downcase.pluralize}"
     action = 'edit'
-    reply_links = authorize_for(controller, action)
+    reply_links = authorize_for(:work_packages, action)
```
It might be a problem having just work_packages here, did this previously work w/ wikis and news etc.?

ulferts committed:
generalize work_package#preview into journals#preview

WorkPackages#preview was essentially two methods now. I could have changed that by splitting it up into #preview and #journal_preview but other journals might need previewing, too.

ulferts committed:
indentation fixes

ulferts committed:
cleanup work_package#quoted

ulferts committed:
removes unsupported format

ulferts committed:
removes hard dependency on work_packages from journals_helper

ulferts committed:
replaces issues controller links on my page

ulferts committed:
fixes context menu link on my page

ulferts committed:
uses work_packages_controller for action menu

It was not yet possible to replace bulk_edit as the controller does not yet provide such an action

ulferts committed:
fixes context menu translations

ulferts committed:
replaces issue links in mails with work package links

apfelfabrik commented:
~~Also, doesn't merge.~~

ulferts commented:
I rebased the branch onto the current feature/rails3. In addition, I:
- fixed what @apfelfabrik mentioned.
- fixed the context menu translations (https://www.openproject.org/issues/695)
- replaced links to issues in the context menu
- replaced links to issues in the mails

ulferts committed:
renames occurances of link_to_issue in favor of link_to_work_package

ulferts commented:
all is green on the internal ci
